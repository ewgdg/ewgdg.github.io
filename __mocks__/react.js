const React = jest.requireActual("react")
module.exports = {
  ...React,
  useContext: jest.fn().mockImplementation(React.useContext),
}
