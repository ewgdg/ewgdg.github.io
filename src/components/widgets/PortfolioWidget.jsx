/* eslint-disable react/no-array-index-key */
/* eslint-disable react/jsx-props-no-spreading */
import React from "react"
import { useStaticQuery, graphql } from "gatsby"
import Container from "@material-ui/core/Container"

import ImageBasedCard from "../thumbnail/ImageBasedCard"
import CardDivision from "../thumbnail/CardDivision"
import AnimatedTitle from "../titles/AnimatedTitle"

import FlexContainer from "../sections/FlexContainer"
import useFlattenMarkdownData from "../others/useFlattenMarkdownData"
import SlideInSection from "../sections/SlideInSection"
// import "../../queries/postsQueries" //no need, gatsby export it to global space

function PortfolioPreview() {
  const { allMarkdownRemark } = useStaticQuery(graphql`
    query {
      allMarkdownRemark(
        sort: { order: DESC, fields: [frontmatter___date] }
        limit: 2
        filter: {
          frontmatter: { isPortfolio: { eq: true }, featuredPost: { eq: true } }
        }
      ) {
        ...PostsFragment
      }
    }
  `)
  const flatten = useFlattenMarkdownData(allMarkdownRemark)

  return (
    <FlexContainer>
      <Container style={{ maxHeight: "100%" }}>
        <div style={{ textAlign: "center" }}>
          <AnimatedTitle title="Games" />
          <p>I enjoy gaming, and I made myself some small games:</p>
        </div>
        <SlideInSection>
          <CardDivision>
            {flatten.map((postProps, i) => (
              <ImageBasedCard key={i} {...postProps} />
            ))}
          </CardDivision>
        </SlideInSection>
      </Container>
    </FlexContainer>
  )
}

export default PortfolioPreview
