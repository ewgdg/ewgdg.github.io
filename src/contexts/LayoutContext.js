import React from "react"

const contextValueRef = { current: { scrollLayer: null, historyState: {} } }
const context = React.createContext(contextValueRef.current)
export default context
export { contextValueRef }
