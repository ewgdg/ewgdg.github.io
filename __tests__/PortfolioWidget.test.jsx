import React from "react"
import { render, getByTestId } from "@testing-library/react"
import { useStaticQuery } from "gatsby"
import PortfolioWidget from "../src/components/widgets/PortfolioWidget"
import LayoutContext from "../src/contexts/LayoutContext"

const { useContext } = jest.requireActual("react")
describe("PortfolioWidget", () => {
  beforeAll(() => {
    // fake post data
    useStaticQuery.mockReturnValue({
      allMarkdownRemark: {
        edges: [
          {
            post: {
              excerpt: "excerpt",
              id: "id",
              fields: {
                slug: "slug",
              },
              frontmatter: {
                title: "title",
                description: "description",
                templateKey: "templateKey",
                externalLink: "localhost",
                date: "OCT 09, 2009",
                tags: "tags",
                featuredPost: false,
                featuredImage: undefined,
              },
            },
          },
        ],
      },
    })

    // set context
    const div = document.createElement("div")
    document.body.appendChild(div)
    div.setAttribute("data-testid", "scrollLayer")
    // const { getByTestId } = render(<div data-testid="scrollLayer" />)
    React.useContext.mockImplementation(context => {
      if (Object.is(LayoutContext, context)) {
        return {
          scrollLayer: div,
        }
      }
      return useContext(context)
    })
  })
  it("should render properly", () => {
    const { container } = render(<PortfolioWidget />, {
      container: getByTestId(document.body, "scrollLayer"),
    })
    expect(container.firstChild).toMatchSnapshot()
  })
})
