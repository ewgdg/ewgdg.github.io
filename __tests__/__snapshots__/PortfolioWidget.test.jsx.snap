// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioWidget should render properly 1`] = `
<div
  class="makeStyles-flexContainer-1 undefined"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
    style="max-height: 100%;"
  >
    <div
      style="text-align: center;"
    >
      <h2
        style="text-align: center;"
      >
        <strong>
          Games
        </strong>
      </h2>
      <p>
        I enjoy gaming, and I made myself some small games:
      </p>
    </div>
    <div
      class="slide-in"
      style="opacity: 0.09999999999999998; transform: matrix(1,0,0,1,0,50);"
    >
      <div
        style="display: inline-block; width: 100%; margin: 0px; height: 100%;"
      >
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-justify-xs-space-around"
          style="height: 100%;"
        >
          <div
            class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5"
            style="height: 400px;"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-112 MuiPaper-rounded"
            >
              <button
                class="MuiButtonBase-root MuiCardActionArea-root makeStyles-action-114"
                tabindex="0"
                type="button"
              >
                <div
                  class="MuiGrid-root makeStyles-media-113 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                  style="background-color: white;"
                  title="title"
                >
                  <div
                    style="margin: 5%;"
                  >
                    <h1>
                      <strong>
                        title
                      </strong>
                    </h1>
                  </div>
                </div>
                <div
                  class="MuiGrid-root makeStyles-content-115 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
                >
                  <div
                    class="MuiCardContent-root"
                  >
                    <h2
                      class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
                    >
                      title
                    </h2>
                    <p
                      class="MuiTypography-root MuiTypography-body2"
                      style="color: rgba(255, 255, 255, 0.8);"
                    >
                      description
                    </p>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
